core/blockchain.go

```diff
@@ -1302,9 +1310,21 @@ func (self *BlockChain) WriteBlock(block *types.Block) (status WriteStatus, err
 	externTd := new(big.Int).Add(block.Difficulty(), ptd)
 
 	// If the total difficulty is higher than our known, add it to the canonical chain
-	// Second clause in the if statement reduces the vulnerability to selfish mining.
-	// Please refer to http://www.cs.cornell.edu/~ie53/publications/btcProcFC.pdf
-	if externTd.Cmp(localTd) > 0 || (externTd.Cmp(localTd) == 0 && mrand.Float64() < 0.5) {
+	// Compare local vs external difficulties
+	tdCompare := externTd.Cmp(localTd)
+
+	// Initialize reorg if incoming td is greater than local.
+	reorg := tdCompare > 0
+
+	// If difficulties are the same, check block numbers.
```

Comment:
I don't think this work out in Proof of Work. If total difficulty is the same, then two chains should be equally preferable. Otherwise, an attacker (with just around 50% hashrate) can craft blocks that is about the same total difficulty but with more blocks that create chain split among newer and older version of geth.

i.e. This line potentially creates a fork.

Reply:
So my suggestion is to remove the `nCompare` check.

Reply:
Referring to this line: https://github.com/ethereumproject/go-ethereum/pull/417/files/9fab3db0e8648d17ff5195ff13b6b2982726c054#diff-30184e52dfee0fbd0ba21f92638b2ec1R1321